File: src/munger.py
#!/usr/bin/python3.4

import io
import sys

ko = None
insymbols = False
with open("kernel.map") as kf: 
    kls = kf.readlines()
    for kl in kls:
        if insymbols:
            syms = kl.split()
            if len(syms) == 3 and (syms[2][0] == "_" or syms[2][-1] == "_"):
                print("Found ", syms[2])
                csym = syms[2].lstrip("_").rstrip("_").upper()
                cline ="#define K_" + csym + " " + "0x" + syms[1] 
                print(cline, file=ko)
        if kl[0:3] == "-- " and kl[3:10] == "Symbols":
            print("Found symbols")
            insymbols = True
            ko = open("kernel.h", "wt")
    if (ko != None):
        ko.close()

File: src/munger.pl
#!/usr/bin/perl
# Perl 3 compatible version for MS-DOS

# Check if filename was provided

$filename = "KERNEL.MAP";

# Open the file
open(FH, $filename) || die "Cannot open file '$filename': $!\n";

# Process each line
while (<FH>) {
    # Match pattern: two hex values and a symbol
    # Adjust the regex pattern based on your specific format
    if (/\b([0-9A-Fa-f][0-9A-Fa-f])\s+([0-9A-Fa-f][0-9A-Fa-f])\s+(\w+)/) {
        $hex1 = $1;
        $hex2 = $2;
        $symbol = $3;

	chop($symbol);
       	$symbol =~ tr/a-z/A-Z/;
	$hex1 =~ tr/a-f/A-F/;
	 
        # Print as #define with symbol and first hex value
        print "#define K_$symbol 0x$hex1\n";
    }
}

close(FH);
